veps/sig-compute/node-pressure-eviction.md

```diff
@@ -0,0 +1,229 @@
+# Overview
+
+Kubernetes has two types of pod eviction.
+
+[API-Initiated Eviction](https://kubernetes.io/docs/concepts/scheduling-eviction/api-eviction/) and [Node Pressure Eviction](https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/). The API-Initiated Eviction is coordinated at the cluster level. This is the kind of eviction used during node draining events and it’s the type of eviction that observes Pod Disruption Budgets.
+
+Node Pressure Eviction is completely independent from the API-Initiated Eviction. This eviction occurs directly on the node driven by the kubelet. Pod Disruption Budgets are not observed and neither is the pod’s termination grace period. This means when a node is under pressure, the kubelet has the ability to kill workloads independently of any cluster wide rules in order to free up resources.
+
+There are ways to influence and even disable Node Pressure Eviction using the kubelet config, but be aware that these are node level configurations and not directly influenced by the workload itself. **Meaning, unlike API-Initiated Eviction, workload owners cannot directly influence how Node Pressure Eviction behaves for their workload.**
+
+## Motivation
+
+Cluster admins often have two initiatives that conflict with one another. On one hand they want to maximise VM workload density within their cluster to reduce cost. On the other hand they want to minimize disruption to those workloads. Having the ability to automate relieving node pressure by live migrating VMs helps with both of these initiatives.
+
+## Goals
+Currently, the VMI api’s EvictionStrategy tunable only works with API-Initiated Eviction. The goal is to extend the behavior of the EvictionStrategy to Node Pressure Eviction as well. This means if a VMI has an EvictionStrategy of LiveMigrate, we expect the system to attempt to live migrate the VMI regardless of which eviction method initiated the eviction.
+
+## Non Goals
+* There is no expectation that NodePressure and API-Initiated eviction will provide the same guarantees.
+
+## Repos
+
+- kubevirt/kubevirt - this is where the majority of the node pressure eviction logic will be implemented
+- kubevirt/hyperconverged-operator - this component will need to expose enabling the feature gate for node pressure eviction handling
+
+## Definition of Users
+- Cluster Admins who wish to influence at the cluster level how VMs behave during Node Pressure Eviction
```

Comment:
I would provide generic role that wants to achieve density on cluster but wants to have sane and safe behavior in case of spikes?